ops/framework.py

```diff
@@ -381,11 +436,25 @@ class Object(metaclass=_Metaclass):
     been created.
 
     """
-    framework = None  # type: Framework
-    handle = None  # type: Handle
     handle_kind = HandleKind()  # type: str
 
-    def __init__(self, parent, key):
+    if TYPE_CHECKING:
+        # to help the type checker and IDEs:
+        # all these are guaranteed to be set at runtime.
+        @property
+        def framework(self) -> 'Framework': ...  # noqa
+        @framework.setter
+        def framework(self, val: 'Framework'): ...  # noqa
+        @property
+        def handle(self) -> 'Handle': ...  # noqa
+        @handle.setter
+        def handle(self, val: 'Handle'): ...  # noqa
```

Comment:
It works; it seems that `on` is the exception here though. It being python3.5, I cannot type without assigning, and `on` I cannot assign to None at init time because I'd override the metaclass-provided instance.
That's probably the reason why I started with that if TYPE_CHECKING thing in class bodies.